### ISSUE TITLE
Compilation errors with dbt-core v1.5.0

### ISSUE THREAD

**Issue by aleix-cd** [get-select/dbt-snowflake-monitoring#102]
Hello there! 👋 

With the latest dbt version upgrade (v1.5.0), we've seen that the `dbt-snowflake-query-tags` package throws compilation errors:
```
Runtime Error
  Compilation Error in test dbt_utils_accepted_range_stg_budget_new_arr__0 (models/company_basics/staging/stg_company_basics.yml)
    Object of type Undefined is not JSON serializable
    
    > in macro get_query_comment (macros/query_comment.sql)
    > called by test dbt_utils_accepted_range_stg_budget_new_arr__0 (models/company_basics/staging/stg_company_basics.yml)
```
☝️ this is just one example, and not the only node. This is happening on all nodes of the dbt project.

I assume this is indicating that there's a value in the `comment_dict` dictionary that is of type _Undefined_.

We've disabled the `dbt-snowflake-monitoring` from our `packages.yml`, and everything compiled correctly, so we are assuming that the problem is somewhere within the `get_query_comment` macro.

**Comment by ian-whitestone** [comment]
Thanks for raising @aleix-cd. We'll keep you posted once we find out what is happening and make a fix!

**Comment by mbkupfer** [comment]
@aleix-cd how did you disable the package. For me, I can't just uninstall as there are still references to it in our codebase that I'd need to refactor since I'm getting a compilation error.

**Comment by ian-whitestone** [comment]
@aleix-cd this has been fixed in the latest release 4.1.1

https://github.com/get-select/dbt-snowflake-query-tags/pull/11/files

**Comment by aleix-cd** [comment]
> @aleix-cd this has been fixed in the latest release 4.1.1
> 
> https://github.com/get-select/dbt-snowflake-query-tags/pull/11/files

Thanks a lot for this, @ian-whitestone! 👍
